Sources/_CryptoExtras/RSA/RSA_boring.swift

```diff
@@ -315,6 +344,215 @@ extension BoringSSLRSAPublicKey {
             return output
         }
 
+        fileprivate func blind<H: HashFunction>(
+            _ message: _RSA.BlindSigning.PreparedMessage,
+            parameters: _RSA.BlindSigning.Parameters<H>
+        ) throws -> (blindedMessage: Data, blindInverse: _RSA.BlindSigning.BlindInverse) {
+            /// ```
+            /// All BN_CTX_get() calls must be made before calling any other functions that use the ctx as an argument.
+            /// ...
+            /// BN_CTX_get() returns a pointer to the BIGNUM, or NULL on error. Once BN_CTX_get() has failed, the
+            /// subsequent calls will return NULL as well, so it is sufficient to check the return value of the last
+            /// BN_CTX_get() call.
+            /// ```
+            /// —— Extract from `man 3 BN_CTX_get`.
+            let bnCtx = CCryptoBoringSSL_BN_CTX_new()
+            CCryptoBoringSSL_BN_CTX_start(bnCtx)
+            defer {
+                CCryptoBoringSSL_BN_CTX_end(bnCtx)
+                CCryptoBoringSSL_BN_CTX_free(bnCtx)
+            }
+            let m = CCryptoBoringSSL_BN_CTX_get(bnCtx)
+            let gcd = CCryptoBoringSSL_BN_CTX_get(bnCtx)
+            let r = CCryptoBoringSSL_BN_CTX_get(bnCtx)
+            let inv = CCryptoBoringSSL_BN_CTX_get(bnCtx)
+            let x = CCryptoBoringSSL_BN_CTX_get(bnCtx)
+            let z = CCryptoBoringSSL_BN_CTX_get(bnCtx)
+            guard z != nil else { throw CryptoKitError.internalBoringSSLError() }
+
+            let rsaPublicKey = CCryptoBoringSSL_EVP_PKEY_get0_RSA(self.pointer)
+            let outputSize = Int(CCryptoBoringSSL_RSA_size(rsaPublicKey))
+
+            // 1. encoded_msg = EMSA-PSS-ENCODE(msg, bit_len(n)) with Hash, MGF, and salt_len as defined in the parameters
+            // 2. If EMSA-PSS-ENCODE raises an error, re-raise the error and stop
+            let hashDigestType = try! DigestType(forDigestType: H.Digest.self)
+            let hash = H.hash(data: message.rawRepresentation)
+            let saltLength: Int32
+            switch parameters.padding {
+            case .PSS: saltLength = Int32(H.Digest.byteCount)
+            case .PSSZERO: saltLength = 0
+            }
+            var encodedMessage = [UInt8](repeating: 0, count: outputSize)
```

Comment:
We could use `Array(unsafeUninitializedCapacity:initializingWith:)` instead of zeroing everything out first.

Reply:
Oh, in fact we don't even need that: we can use `withUnsafeTemporaryAllocation` to avoid the heap allocation altogether.